### PR TITLE
feat: allow transforming when virtual module ids are used

### PR DESCRIPTION
**Pull request by klipstein**
replaces #1

### CODE REVIEW

File: src/index.ts [hunk 1]
```diff
@@ -18,11 +18,11 @@ export type PluginOptions = {
   tsconfig?: string | false
 } & Pick<SwcConfig, Exclude<keyof SwcConfig, 'filename'>>;
 
-const INCLUDE_REGEXP = /\.[jt]sx?$/;
+const INCLUDE_REGEXP = /\.m?[jt]sx?$/;
 const EXCLUDE_REGEXP = /node_modules/;
 
 const resolveFile = (resolved: string, index = false) => {
-  for (const ext of ['.ts', '.js', '.tsx', '.jsx']) {
+  for (const ext of ['.ts', '.js', '.tsx', '.jsx', '.mjs']) {
```
Comment (SukkaW) [inline src/index.ts]:
```suggestion
  for (const ext of ['.ts', '.mjs', '.js', '.tsx', '.jsx']) {
```

`.mjs` should have higher priority than `.js`.

**Review by SukkaW — CHANGES_REQUESTED** [review]

**Comment by SukkaW** [comment]
I have submitted a bug report at swc about `\0`: https://github.com/swc-project/swc/issues/2853

BTW, would you mind splitting your PR? The PR has introduced two different changes.

**Comment by klipstein** [comment]
> I have submitted a bug report at swc about `\0`: [swc-project/swc#2853](https://github.com/swc-project/swc/issues/2853)

Definitely better to fix it upstream.

> BTW, would you mind splitting your PR? The PR has introduced two different changes.

I'll split the pull.

File: src/index.ts (hunk 1 above)
Comment (klipstein) [reply src/index.ts]: Fixed it in a separate pull: https://github.com/SukkaW/rollup-plugin-swc/pull/3

**Comment by klipstein** [comment]
> > BTW, would you mind splitting your PR? The PR has introduced two different changes.
> 
> I'll split the pull.

This pull now only contains the fix for the virtual module ids. And #3 provides the mjs feature.

File: src/index.ts [hunk 2]
```diff
@@ -104,7 +104,18 @@ function swc(options: PluginOptions = {}): Plugin {
         }
       ]);
 
-      return swcTransform(code, swcOption);
+      // swc cannot transform module ids with "\0" (would throw
+      // "error: Unterminated string constant"), which can be used by
+      // some plugins (e.g. "@rollup/plugin-commonjs"), see:
+      // - https://rollupjs.org/guide/en/#conventions
+      // - https://github.com/rollup/plugins/blob/02fb349d315f0ffc55970fba5de20e23f8ead881/packages/commonjs/src/helpers.js#L15
+      return swcTransform(code.replace('\0', '\\0'), swcOption)
```
Comment (SukkaW) [inline src/index.ts]:
```suggestion
      return swcTransform(code.replaceAll('\0', '\\0'), swcOption)
```

Use ~~`String.prototype.replaceAll`~~ (Do not use `replaceAll` as the plugin targets `Node.js >= 12`) or `/\0/g`. There might be more than one virtual module being imported, maybe by other rollup plugins.

Also, `\\0` is kinda dangerous: maybe someone would use `\\0` in their code, and the plugin might unexpectedly unescape it after swc transformation. I'd prefer something like `$$__SECRET_SLASH_0_ESCAPE_IDENTIFIER_DO_NOT_USE_OR_YOU_WILL_BE_FIRED__$$`

Comment (klipstein) [reply src/index.ts]: You are right. Did not spend further thoughts on that change after you opened the bug report at swc. Will adjust it as suggested.

Comment (klipstein) [reply src/index.ts]: Applied the changes here: https://github.com/SukkaW/rollup-plugin-swc/pull/2/commits/9bf7b4e170c6240e109313b929f2f0ca192e0bc2

**Review by SukkaW — APPROVED** [review]

**Pull request merged**